### PR TITLE
Fix BooleanMatch and BooleanNotEqualMatch doing Boolean comparison.

### LINKED ISSUE #11156: [Bug] Boolean match bug
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

The class of BooleanMath and BooleanNotEqualMatch，the first compare method.
It's almost always a mistake to compare two instances of java.lang.String or boxed types like java.lang.Integer using reference equality == or !=, because it is not comparing actual value but locations in memory.

### What you expected to happen

use equals() not == or !=

### How to reproduce

Update methods, using the equals api

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wu-sheng**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #11156.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

I believe codes can't run into this part, but fix it anyway as reported.

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/BooleanMatch.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 @BooleanValueFilterMatcher
 public class BooleanMatch {
     public boolean match(Boolean left, Boolean right) {
-        return left == right;
+        return left.equals(right);
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/BooleanMatch.java:26]: When `left == null`, NPE is thrown

```suggestion
        return Objects.equals(left, right);
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/BooleanNotEqualMatch.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 @BooleanValueFilterMatcher
 public class BooleanNotEqualMatch {
     public boolean match(Boolean left, Boolean right) {
-        return left != right;
+        return !left.equals(right);
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/expression/BooleanNotEqualMatch.java:26]:
```suggestion
        return !Objects.equals(left, right);
```

**Review by kezhenxu94 — APPROVED** [review]
you might need to add `import` after applying the suggestions

**Review by wankai123 — APPROVED** [review]

**Comment by wu-sheng** [comment]
I don't think it should be null? Actually, I don't know how codes land here.

**Comment by kezhenxu94** [comment]
> I don't think it should be null? Actually, I don't know how codes land here.

`Boolean` can be null, we have a method `match(boolean left, boolean right)` that `left` and `right` must not be null, and the one you modified `match(Boolean left, Boolean right)` is for those parameters that might be null

**Comment by wu-sheng** [comment]
I mean for OAL context, it seems can't execute here. This is just an in-case method.

**Comment by wu-sheng** [comment]
The only reason I changed, it is from an issue from code scaning.

**Comment by kezhenxu94** [comment]
> The only reason I changed, it is from an issue from code scaning.

Code scanning absolutely will report potential NPE after your change.

**Comment by wu-sheng** [comment]
I don't care about that. This fix is only because the codes are wrong. Potential npe doesn't mean wrong.

**Pull request merged**